engine/class_modules/paladin/sc_paladin.cpp

```diff
@@ -2418,12 +2532,31 @@ double paladin_t::resource_loss( resource_e resource_type, double amount, gain_t
   if (
       resource_type == RESOURCE_HOLY_POWER &&
       result > 0 &&
-      legendary.from_dusk_till_dawn -> ok() &&
-      resources.current[ RESOURCE_HOLY_POWER ] == legendary.from_dusk_till_dawn -> effectN( 2 ).base_value()
+      legendary.of_dusk_and_dawn -> ok() &&
+      resources.current[ RESOURCE_HOLY_POWER ] == legendary.of_dusk_and_dawn -> effectN( 2 ).base_value()
     )
   {
     buffs.blessing_of_dusk -> trigger();
   }
+
+  if (
+      resource_type == RESOURCE_HEALTH &&
+      legendary.reign_of_endless_kings -> ok() &&
+      ! buffs.reign_of_ancient_kings -> up() &&
```

Comment:
I'd rename this buff to make it clearer that it models the internal cooldown

Reply:
You could also just use a cooldown_t object instead, since you don't need the buff but only its duration

Reply:
But that's what it's called ingame D:

Reply:
I don't see how obfuscating the icd to just be a hidden trigger is cleaner than using the in-game functionality.

Reply:
Mostly because I think that it'd be confusing for the user. Seeing a buff simply called "Reign of Endless Kings" I think isn't clear enough that it represents the ICD (hence my first suggestion to rename it). 
My second suggestion is because I don't think having a visible buff for it is necessary and a simple cooldown is lighter to implement than a full buff.